# Task
Load the datasets "sitc_product_year_4.csv", "umap_layout_hs92.csv", "top_edges_hs92.csv", "product_sitc.csv", and "hs92_product_year_4.csv", then display their head, info, and descriptive statistics, summarize their content, and suggest potential visualization strategies.

## Load sitc_product_year_4.csv

### Subtask:
Load the 'sitc_product_year_4.csv' file into a pandas DataFrame and display its head, info, and descriptive statistics to understand its structure and content.


**Reasoning**:
To load the specified CSV file into a pandas DataFrame and then display its head, info, and descriptive statistics, I will use a single code block. This will fulfill all the instructions given in the subtask.



In [1]:
data_folder = "d:/download/hs_data/"

In [1]:
import pandas as pd

data_folder = '/Users/nadira/Documents/Work/cottonborry/data/'  # Make sure this ends with /

df_sitc_product_year_4 = pd.read_csv(data_folder + 'sitc_product_year_4.csv')

print("DataFrame Head:")
print(df_sitc_product_year_4.head())

print("\nDataFrame Info:")
df_sitc_product_year_4.info()

print("\nDataFrame Descriptive Statistics:")
print(df_sitc_product_year_4.describe())

DataFrame Head:
   product_id product_sitc_code  year  export_value  import_value    pci
0         650              0011  1962     401238525     401238525 -1.542
1         651              0012  1962      39010957      39010957 -1.710
2         652              0013  1962      56963680      56963680 -0.671
3         653              0014  1962      21271244      21271244 -0.846
4         654              0015  1962      88023339      88023339 -0.455

DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48708 entries, 0 to 48707
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   product_id         48708 non-null  int64  
 1   product_sitc_code  48708 non-null  object 
 2   year               48708 non-null  int64  
 3   export_value       48708 non-null  int64  
 4   import_value       48708 non-null  int64  
 5   pci                48708 non-null  float64
dtypes: float64(1), int64(4), object(

## Load umap_layout_hs92.csv

### Subtask:
Load the 'umap_layout_hs92.csv' file into a pandas DataFrame and display its head, info, and descriptive statistics.


**Reasoning**:
To load the specified CSV file into a pandas DataFrame and then display its head, info, and descriptive statistics, I will use a single code block. This will fulfill all the instructions given in the subtask.



In [3]:
df_umap_layout_hs92 = pd.read_csv(data_folder+'umap_layout_hs92.csv')

print("DataFrame Head:")
print(df_umap_layout_hs92.head())

print("\nDataFrame Info:")
df_umap_layout_hs92.info()

print("\nDataFrame Descriptive Statistics:")
print(df_umap_layout_hs92.describe())

DataFrame Head:
   product_hs92_code  product_space_x  product_space_y  \
0               8518        -1.496793        -1.784722   
1               8535        -1.165153        -1.164215   
2               9608        -0.617562        -1.376214   
3               1519         0.352759         0.930167   
4               6805        -0.695376        -0.126305   

                        product_space_cluster_name  
0                  Electronic and Electrical Goods  
1  Metalworking and Electrical Machinery and Parts  
2                           Textile and Home Goods  
3                                      Agriculture  
4                           Textile and Home Goods  

DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 865 entries, 0 to 864
Data columns (total 4 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_hs92_code           865 non-null    int64  
 1   product_space_x     

## Load top_edges_hs92.csv

### Subtask:
Load the 'top_edges_hs92.csv' file into a pandas DataFrame and display its head, info, and descriptive statistics.

**Reasoning**:
To load the specified CSV file into a pandas DataFrame and then display its head, info, and descriptive statistics, I will use a single code block. This will fulfill all the instructions given in the subtask.



In [5]:
df_top_edges_hs92 = pd.read_csv(data_folder+'top_edges_hs92.csv')

print("DataFrame Head:")
print(df_top_edges_hs92.head())

print("\nDataFrame Info:")
df_top_edges_hs92.info()

print("\nDataFrame Descriptive Statistics:")
print(df_top_edges_hs92.describe())

DataFrame Head:
   product_hs92_code_source  product_hs92_code_target
0                      6110                      6104
1                      6104                      6110
2                      6106                      6104
3                      6104                      6106
4                      6108                      6107

DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4316 entries, 0 to 4315
Data columns (total 2 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   product_hs92_code_source  4316 non-null   int64
 1   product_hs92_code_target  4316 non-null   int64
dtypes: int64(2)
memory usage: 67.6 KB

DataFrame Descriptive Statistics:
       product_hs92_code_source  product_hs92_code_target
count               4316.000000               4316.000000
mean                5426.596849               5426.596849
std                 2803.030447               2803.030447
min                

## Load product_sitc.csv

### Subtask:
Load the 'product_sitc.csv' file into a pandas DataFrame and display its head, info, and descriptive statistics.

**Reasoning**:
To load the specified CSV file into a pandas DataFrame and then display its head, info, and descriptive statistics, I will use a single code block. This will fulfill all the instructions given in the subtask.



In [7]:
df_product_sitc = pd.read_csv(data_folder+'product_sitc.csv')

print("DataFrame Head:")
print(df_product_sitc.head())

print("\nDataFrame Info:")
df_product_sitc.info()

print("\nDataFrame Descriptive Statistics:")
print(df_product_sitc.describe())

DataFrame Head:
   product_id product_sitc_code  product_level  \
0           1                 0              1   
1           2                 1              1   
2           3                 2              1   
3           4                 3              1   
4           5                 4              1   

                                      product_name product_name_short  \
0           Food and live animals chiefly for food               Food   
1                            Beverages and tobacco          Beverages   
2          Crude materials, inedible, except fuels    Crude materials   
3  Mineral fuels, lubricants and related materials              Fuels   
4        Animal and vegetable oils, fats and waxes     Vegetable oils   

   product_parent_id product_id_hierarchy  show_feasibility natural_resource  
0                NaN                    1              True              NaN  
1                NaN                    2              True              NaN  
2      

## Load hs92_product_year_4.csv

### Subtask:
Load the 'hs92_product_year_4.csv' file into a pandas DataFrame and display its head, info, and descriptive statistics.

**Reasoning**:
To load the specified CSV file into a pandas DataFrame and then display its head, info, and descriptive statistics, I will use a single code block. This will fulfill all the instructions given in the subtask.



In [9]:
df_hs92_product_year_4 = pd.read_csv(data_folder+'hs92_product_year_4.csv')

print("DataFrame Head:")
print(df_hs92_product_year_4.head())

print("\nDataFrame Info:")
df_hs92_product_year_4.info()

print("\nDataFrame Descriptive Statistics:")
print(df_hs92_product_year_4.describe())

DataFrame Head:
   product_id product_hs92_code  year  export_value  import_value    pci
0         650              0101  1995    1018155630    1018155630  0.044
1         651              0102  1995    5208104239    5208104239 -0.457
2         652              0103  1995    1586684113    1586684113  0.024
3         653              0104  1995     866772784     866772784 -1.035
4         654              0105  1995     927028221     927028221 -0.168

DataFrame Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35897 entries, 0 to 35896
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   product_id         35897 non-null  int64  
 1   product_hs92_code  35897 non-null  object 
 2   year               35897 non-null  int64  
 3   export_value       35897 non-null  int64  
 4   import_value       35897 non-null  int64  
 5   pci                35897 non-null  float64
dtypes: float64(1), int64(4), object(

## Summary of DataFrames

### `df_sitc_product_year_4`

This DataFrame contains data related to products categorized by SITC (Standard International Trade Classification) codes across different years. It includes:

- `product_id`: A unique identifier for each product.
- `product_sitc_code`: The SITC code for the product, indicating its classification.
- `year`: The year of the trade data.
- `export_value`: The total export value for the product in that year.
- `import_value`: The total import value for the product in that year.
- `pci`: A Product Complexity Index (PCI) score, likely indicating the complexity of the product.

**Observations:**
- There are `48708` entries, covering years from `1962` to `2023`.
- `export_value` and `import_value` have a wide range, indicating varying trade volumes for different products and years.
- `pci` values range from approximately `-4.3` to `2.7`, suggesting a spectrum of product complexity.
- `product_sitc_code` is an object type, implying it might contain alphanumeric codes or categorical data.



### `df_umap_layout_hs92`

This DataFrame provides UMAP layout coordinates and cluster information for products classified by HS92 codes. It includes:

- `product_hs92_code`: The HS92 code for the product.
- `product_space_x`: The x-coordinate of the product in the UMAP layout, representing a dimension in the product space.
- `product_space_y`: The y-coordinate of the product in the UMAP layout, representing another dimension in the product space.
- `product_space_cluster_name`: The name of the cluster to which the product belongs, providing a categorical grouping.

**Observations:**
- There are `865` entries, indicating UMAP coordinates and cluster assignments for 865 HS92 products.
- `product_hs92_code` is an integer type, likely representing unique product codes.
- `product_space_x` and `product_space_y` are float types, ranging from approximately `-2.9` to `3.2` and `-2.1` to `2.0` respectively, defining the product positions in a 2D space.
- `product_space_cluster_name` is an object type, signifying categorical cluster names, which could be useful for visualization and grouping.

### `df_top_edges_hs92`

This DataFrame describes the top edges between products based on their HS92 codes, likely representing relationships or transitions in product space. It includes:

- `product_hs92_code_source`: The HS92 code of the source product in an edge relationship.
- `product_hs92_code_target`: The HS92 code of the target product in an edge relationship.

**Observations:**
- There are `4316` entries, indicating a substantial number of identified relationships between products.
- Both `product_hs92_code_source` and `product_hs92_code_target` are integer types, representing the unique identifiers of products.
- The descriptive statistics show that the product codes range from `101` to `9706`, suggesting a wide range of products are involved in these relationships.

### `df_product_sitc`

This DataFrame provides detailed information about products classified by SITC codes. It includes:

- `product_id`: A unique identifier for each product.
- `product_sitc_code`: The SITC code for the product.
- `product_level`: The hierarchical level of the product in the SITC classification.
- `product_name`: The full name of the product.
- `product_name_short`: A shortened name for the product.
- `product_parent_id`: The `product_id` of the parent product in the hierarchy (can be NaN).
- `product_id_hierarchy`: A hierarchical representation of the product ID.
- `show_feasibility`: A boolean indicating whether the product's feasibility should be shown.
- `natural_resource`: Indicates if the product is a natural resource (can be NaN).

**Observations:**
- There are `870` entries, detailing a comprehensive list of SITC products.
- `product_id` and `product_level` are integer types. `product_level` ranges from `1` to `4`, indicating different levels of classification granularity.
- `product_sitc_code`, `product_name`, `product_name_short`, and `product_id_hierarchy` are object types, containing textual or mixed data.
- `product_parent_id` has `860` non-null values out of `870`, suggesting some products might be top-level or their parent information is missing.
- `natural_resource` has `789` non-null values, indicating that this information is not available for all products.
- `show_feasibility` is a boolean type, consistently `True` for all entries, suggesting it might be a general flag rather than a product-specific one.

### `df_hs92_product_year_4`

This DataFrame contains data related to products categorized by HS92 (Harmonized System 1992) codes across different years. It includes:

- `product_id`: A unique identifier for each product.
- `product_hs92_code`: The HS92 code for the product, indicating its classification.
- `year`: The year of the trade data.
- `export_value`: The total export value for the product in that year.
- `import_value`: The total import value for the product in that year.
- `pci`: A Product Complexity Index (PCI) score, similar to `df_sitc_product_year_4`, but based on HS92 classification.

**Observations:**
- There are `35897` entries, covering years from `1995` to `2023`.
- `export_value` and `import_value` show a broad range of trade activity.
- `pci` values range from approximately `-4.7` to `4.0`, indicating varying levels of product complexity within the HS92 classification.
- `product_hs92_code` is an object type, suggesting it might contain alphanumeric codes or be treated as categorical data.

### `df_hs92_product_year_4`

This DataFrame contains data related to products categorized by HS92 (Harmonized System 1992) codes across different years. It includes:

- `product_id`: A unique identifier for each product.
- `product_hs92_code`: The HS92 code for the product, indicating its classification.
- `year`: The year of the trade data.
- `export_value`: The total export value for the product in that year.
- `import_value`: The total import value for the product in that year.
- `pci`: A Product Complexity Index (PCI) score, similar to `df_sitc_product_year_4`, but based on HS92 classification.

**Observations:**
- There are `35897` entries, covering years from `1995` to `2023`.
- `export_value` and `import_value` show a broad range of trade activity.
- `pci` values range from approximately `-4.7` to `4.0`, indicating varying levels of product complexity within the HS92 classification.
- `product_hs92_code` is an object type, suggesting it might contain alphanumeric codes or be treated as categorical data.


## Potential Visualization Strategies

### `df_sitc_product_year_4`

This DataFrame contains time-series data for trade values and product complexity. Potential visualizations include:

- **Line plot of `export_value` and `import_value` over `year`**: To observe trends in trade for specific `product_sitc_code`s or aggregate trade trends across all products.
- **Line plot of `pci` over `year`**: To track changes in product complexity over time.
- **Bar chart of average `export_value` or `import_value` by `product_sitc_code`**: To compare trade volumes across different product categories.
- **Scatter plot of `pci` vs. `export_value` or `import_value`**: To explore potential relationships between product complexity and trade performance, possibly animated over time for dynamic insights.

### `df_umap_layout_hs92`

This DataFrame provides UMAP coordinates for products and their cluster assignments. Potential visualizations include:

- **Scatter plot of `product_space_x` vs. `product_space_y`**: To visualize the product space, where each point represents a product. Points can be colored by `product_space_cluster_name` to highlight distinct clusters.
- **Interactive scatter plot**: Using libraries like Plotly or Altair, to allow users to hover over points and see `product_hs92_code` or `product_space_cluster_name` details.
- **Density plots or contour plots**: Over the `product_space_x` and `product_space_y` dimensions, to show areas of higher product concentration.

### `df_top_edges_hs92`

This DataFrame describes relationships between products. Potential visualizations include:

- **Network graph**: A force-directed graph (e.g., using libraries like NetworkX with Matplotlib, or interactive libraries like Plotly, Bokeh, or D3.js) where nodes are `product_hs92_code`s and edges represent the relationships between source and target products. This can help visualize product interconnectedness and identify central products or communities.
- **Chord diagram**: To show the flow or relationships between different product categories or individual products.
- **Adjacency matrix heatmap**: Representing the connections between products as a heatmap, which can reveal patterns of interconnectedness.

## Final Task

### Subtask:
Summarize the content of `df_product_sitc` and suggest potential visualization strategies.


## Summary:

### Data Analysis Key Findings

*   **`df_sitc_product_year_4`**: This DataFrame contains 48,708 entries across 6 columns, detailing SITC product trade data from 1962 to 2023. It includes `product_id`, `product_sitc_code`, `year`, `export_value`, `import_value`, and `pci`. Export and import values show a wide range, from approximately 6,884 to over 1.38 trillion, indicating significant trade variance. The Product Complexity Index (PCI) ranges from -4.318 to 2.785.
*   **`df_umap_layout_hs92`**: Comprising 865 entries and 4 columns, this DataFrame maps HS92 products into a 2D UMAP space, providing `product_hs92_code`, `product_space_x`, `product_space_y`, and `product_space_cluster_name`. The UMAP coordinates (`x` and `y`) range from approximately -2.9 to 3.2 and -2.1 to 2.0, respectively, defining product positions.
*   **`df_top_edges_hs92`**: This DataFrame contains 4,316 entries describing relationships between HS92 products via `product_hs92_code_source` and `product_hs92_code_target`. Product codes involved in these relationships range from 101 to 9706.
*   **`df_product_sitc`**: With 870 entries and 9 columns, this DataFrame provides detailed information on SITC products, including `product_id`, `product_sitc_code`, `product_level`, `product_name`, and hierarchical information. `product_level` ranges from 1 to 4, indicating classification granularity. There are 10 missing `product_parent_id` values and 81 missing `natural_resource` values.
*   **`df_hs92_product_year_4`**: This DataFrame holds 35,897 entries of HS92 product trade data from 1995 to 2023, with columns similar to `df_sitc_product_year_4` (i.e., `product_id`, `product_hs92_code`, `year`, `export_value`, `import_value`, `pci`). PCI values range from -4.721 to 4.062, capturing varying product complexities within the HS92 classification.

### Insights or Next Steps

*   **Integrate and Analyze Trade Dynamics**: The `_product_year_4` DataFrames (SITC and HS92) offer rich time-series data on trade values and product complexity. Future analysis could focus on identifying trends in `export_value`, `import_value`, and `pci` over time, potentially comparing trade dynamics between SITC and HS92 classifications using line plots and aggregated bar charts.
*   **Visualize Product Relationships and Space**: Utilize the UMAP coordinates from `df_umap_layout_hs92` in a scatter plot (colored by `product_space_cluster_name`) to visually explore product clusters. Concurrently, a network graph constructed from `df_top_edges_hs92` would reveal interconnectedness and central products in the HS92 product space. This combined visualization approach can provide insights into product proximity and transitional pathways.
